tests/unit/test_opentelemetry_tracing.py

```diff
@@ -0,0 +1,215 @@
+    span_processor = SimpleExportSpanProcessor(memory_exporter)
+    tracer_provider.add_span_processor(span_processor)
+    trace.set_tracer_provider(tracer_provider)
+    yield memory_exporter
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_opentelemetry_not_installed(setup):
+    temp_module = sys.modules["opentelemetry"]
+    sys.modules["opentelemetry"] = None
+    reload_module(opentelemetry_tracing)
+    with opentelemetry_tracing.create_span("No-op for opentelemetry") as span:
+        assert span is None
+    sys.modules["opentelemetry"] = temp_module
+    reload_module(opentelemetry_tracing)
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_opentelemetry_success(setup):
+    expected_attributes = {"foo": "baz", "db.system": "BigQuery"}
+
+    with opentelemetry_tracing.create_span(
+        TEST_SPAN_NAME, attributes=TEST_SPAN_ATTRIBUTES, client=None, job_ref=None
+    ) as span:
+        assert span is not None
+        assert span.name == TEST_SPAN_NAME
+        assert span.attributes == expected_attributes
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_default_client_attributes(setup):
+    expected_attributes = {
+        "foo": "baz",
+        "db.system": "BigQuery",
+        "db.name": "test_project",
+        "location": "test_location",
+    }
+    with mock.patch("google.cloud.bigquery.client.Client") as test_client:
+        test_client.project = "test_project"
+        test_client.location = "test_location"
+        with opentelemetry_tracing.create_span(
+            TEST_SPAN_NAME, attributes=TEST_SPAN_ATTRIBUTES, client=test_client
+        ) as span:
+            assert span is not None
+            assert span.name == TEST_SPAN_NAME
+            assert span.attributes == expected_attributes
+
+
+@pytest.mark.skipif(opentelemetry is None, reason="Require `opentelemetry`")
+def test_default_job_attributes(setup):
+    import google.cloud._helpers
+
+    time_created = datetime.datetime(
+        2010, 5, 19, 16, 0, 0, tzinfo=google.cloud._helpers.UTC
+    )
+    started_time = datetime.datetime(
+        2011, 10, 1, 16, 0, 0, tzinfo=google.cloud._helpers.UTC
+    )
+    ended_time = datetime.datetime(
+        2011, 10, 2, 16, 0, 0, tzinfo=google.cloud._helpers.UTC
+    )
+    error_result = [
+        {"errorResult1": "some_error_result1", "errorResult2": "some_error_result2"}
+    ]
+
+    expected_attributes = {
+        "db.system": "BigQuery",
+        "db.name": "test_project_id",
+        "location": "test_location",
+        "num_child_jobs": "0",
+        "job_id": "test_job_id",
+        "foo": "baz",
+        "parent_job_id": "parent_job_id",
+        "timeCreated": time_created.strftime("%d-%b-%Y (%H:%M:%S.%f)"),
+        "timeStarted": started_time.strftime("%d-%b-%Y (%H:%M:%S.%f)"),
+        "timeEnded": ended_time.strftime("%d-%b-%Y (%H:%M:%S.%f)"),
```

Comment:
Do the attributes really use dd-mmm-yyyy formatting instead of [ISO 8601](https://xkcd.com/1179/)? 😢